### PR TITLE
Renaming storage.get_all_buckets to storage.list_buckets.

### PR DESCRIPTION
**Pull request by dhermes**
Also adding support for optional query parameters of `storage.buckets.list`.

Note this rename is "proposed" in #761. If there are issues we don't have to move forward with it.

Relates to #802.

### CODE REVIEW

File: gcloud/storage/demo/demo.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 connection = storage.get_connection()
 
 # OK, now let's look at all of the buckets...
-print(demo.get_all_buckets(connection))  # This might take a second...
+print(demo.list_buckets(connection))  # This might take a second...
```
Comment (tseaver) [inline gcloud/storage/demo/demo.py]: Doesn't this need to be wrapped in `list()`?

File: gcloud/storage/demo/demo.py [hunk 2]
```diff
@@ -34,7 +34,7 @@
 print(bucket)
 
 # Let's look at all of the buckets again...
-print(demo.get_all_buckets(connection))
+print(demo.list_buckets(connection))
```
Comment (tseaver) [inline gcloud/storage/demo/demo.py]: Ditto.

File: gcloud/storage/test_api.py [hunk 3]
```diff
@@ -78,42 +78,46 @@ def test_use_default(self):
...
+    def _callFUT(self, *args, **kwargs):
+        from gcloud.storage.api import list_buckets
+        return list_buckets(*args, **kwargs)
 
     def test_empty(self):
+        from six.moves.urllib.parse import urlencode
         from gcloud.storage.connection import Connection
         PROJECT = 'project'
         conn = Connection()
+        query_params = urlencode({'project': PROJECT, 'projection': 'noAcl'})
```
Comment (tseaver) [inline gcloud/storage/test_api.py]: This may have hash-ordering problems.

Comment (dhermes) [inline gcloud/storage/test_api.py]: I ran into this in `test_all_arguments`, so can hack this into here. Thanks.

File: gcloud/storage/demo/demo.py (hunk 1 above)
Comment (dhermes) [inline gcloud/storage/demo/demo.py]: On it. Going to get it into the original commit.

**Comment by dhermes** [comment]
@tseaver Addressed issues. PTAL.

**Comment by tseaver** [comment]
LGTM

**Pull request merged**